sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoJsonDeserializer.java

```diff
@@ -83,32 +73,30 @@ private static GeoObject read(JsonNode node) {
         GeoBoundingBox boundingBox = readBoundingBox(node);
         Map<String, Object> properties = readProperties(node);
 
-        switch (type) {
-            case POINT_TYPE:
-                return new GeoPoint(readCoordinate(coordinates), boundingBox, properties);
-            case LINE_STRING_TYPE:
-                return new GeoLineString(readCoordinates(coordinates), boundingBox, properties);
-            case POLYGON_TYPE:
-                List<GeoLinearRing> rings = new ArrayList<>();
-                coordinates.forEach(ring -> rings.add(new GeoLinearRing(readCoordinates(ring))));
-
-                return new GeoPolygon(rings, boundingBox, properties);
-            case MULTI_POINT_TYPE:
-                List<GeoPoint> points = new ArrayList<>();
-                readCoordinates(coordinates).forEach(position -> points.add(new GeoPoint(position)));
-
-                return new GeoPointCollection(points, boundingBox, properties);
-            case MULTI_LINE_STRING_TYPE:
-                List<GeoLineString> lines = new ArrayList<>();
-                coordinates.forEach(line -> lines.add(new GeoLineString(readCoordinates(line))));
-
-                return new GeoLineStringCollection(lines, boundingBox, properties);
-            case MULTI_POLYGON_TYPE:
-                return readMultiPolygon(coordinates, boundingBox, properties);
-            default:
-                throw LOGGER.logExceptionAsError(new IllegalStateException(
-                    String.format("Unsupported geo type %s.", type)));
+        if (Objects.equals(type, GeoObjectType.POINT.getJsonType())) {
+            return new GeoPoint(readCoordinate(coordinates), boundingBox, properties);
+        } else if (Objects.equals(type, GeoObjectType.LINE_STRING.getJsonType())) {
+            return new GeoLineString(readCoordinates(coordinates), boundingBox, properties);
+        } else if (Objects.equals(type, GeoObjectType.POLYGON.getJsonType())) {
+            List<GeoLinearRing> rings = new ArrayList<>();
+            coordinates.forEach(ring -> rings.add(new GeoLinearRing(readCoordinates(ring))));
+
+            return new GeoPolygon(rings, boundingBox, properties);
+        } else if (Objects.equals(type, GeoObjectType.MULTI_POINT.getJsonType())) {
+            List<GeoPoint> points = new ArrayList<>();
+            readCoordinates(coordinates).forEach(position -> points.add(new GeoPoint(position)));
+
+            return new GeoPointCollection(points, boundingBox, properties);
+        } else if (Objects.equals(type, GeoObjectType.MULTI_LINE_STRING.getJsonType())) {
+            List<GeoLineString> lines = new ArrayList<>();
+            coordinates.forEach(line -> lines.add(new GeoLineString(readCoordinates(line))));
+
+            return new GeoLineStringCollection(lines, boundingBox, properties);
+        } else if (Objects.equals(type, GeoObjectType.MULTI_POLYGON.getJsonType())) {
+            return readMultiPolygon(coordinates, boundingBox, properties);
```

Comment:
Can we still use the switch case here instead of if-else?

Reply:
It previously was a switch but that requires using constants, so I couldn't use the `getJsonType()` value associated with the `GeoObjectType`. The other options I had used was adding a string -> enum method on `GeoObjectType` but I didn't really like that either.

Do you have a preference on either? FYI, these cases shouldn't change, or at least not often, as they are tightly coupled to the GeoJSON RFC